### DISCUSSION TITLE
pass url parameters to end point "Custom tool"

### DISCUSSION THREAD

**Discussion by remixdance** [FlowiseAI/Flowise#875] [Q&A]
Hi I dont know if is possible to pass a param on the url of my end point that I´m using in my custom tool for example:

const fetch = require('node-fetch');
const url = 'https://myserver.com/api/v2/report?Name=ABC;
const options = {
    method: 'GET',
    headers: {
        'Authorization': 'c',
        'Content-Type': 'application/json'
    }
};
try {
    const response = await fetch(url, options);
    const text = await response.text();
    return text;
} catch (error) {
    console.error(error);
    return '';
}

Where the params it could be Name, ID, or Company etc and this params make dynamically or Do I need to make a custom tool per param? 

Thanks

**Comment by chungyau97** [comment]
Hi @remixdance,

You could try something like this:
![image](https://github.com/FlowiseAI/Flowise/assets/33013947/cdbf48f7-c34d-4eb1-89dc-c8c659eaf3ec)
![image](https://github.com/FlowiseAI/Flowise/assets/33013947/89581719-8800-41d6-9910-2e07f02ba4ce)
![image](https://github.com/FlowiseAI/Flowise/assets/33013947/87d8132c-15a6-4020-9a1d-431448859aee)

Changing the `id` into `queryParam`, then send a message `get result using "Company=ABC"`.

**Reply by remixdance** [reply]
thank so much¡
